adafruit_pybadger/pyportal.py

```diff
@@ -0,0 +1,72 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2020 Kattni Rembor for Adafruit Industries
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""
+`adafruit_pybadger.pyportal`
+================================================================================
+
+Badge-focused CircuitPython helper library for PyPortal.
+
+
+* Author(s): Kattni Rembor
+
+Implementation Notes
+--------------------
+
+**Hardware:**
+
+* `Adafruit PyPortal <https://www.adafruit.com/product/4116>`_
+
+**Software and Dependencies:**
+
+* Adafruit CircuitPython firmware for the supported boards:
+  https://github.com/adafruit/circuitpython/releases
+
+"""
+
+import audioio
+from adafruit_pybadger.pybadger_base import PyBadgerBase
+
+__version__ = "0.0.0-auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_PyBadger.git"
+
+
+class PyPortal(PyBadgerBase):
+    """Class that represents a single PyPortal."""
+
+    _audio_out = audioio.AudioOut
+    _neopixel_count = 1
+
+    @property
+    def _unsupported(self):
+        """This feature is not supported on PyPortal."""
+        raise NotImplementedError("This feature is not supported on PyPortal.")
+
+    # The following is a list of the features available in other PyBadger modules but
+    # not available for PyPortal. If called while using a PyPortal, they will result in the
+    # NotImplementedError raised in the property above.
+    play_file = _unsupported
```

Comment:
Why is play_file unsupported?

Reply:
It wasn't left unsupported for a specific reason. I started the PyPortal class with a copy of the existing CLUE class and it got brought in as unsupported from there. I just didn't have time to work on adding it after the display stuff was working before I went to bed last night. 

I will add some more commits today to add the other functionalities that are possible on this hardware. 

Looking back in the CLUE file now it is setting up an AudioPWMOut object but then setting play_file as unsupported. That seems odd to me at first glance, but maybe there is a reason. I'll poke around a bit today.